Update person table per new _delta vocabulary approach

briangow opened the pull request:
The primary purpose of this PR is to update the build of the person table to make use of the updated vocabulary approach from #37. 

This first requires changes to the build of `lk_pat_race_concept`:
- From the vocabulary, use `concept_name` instead of `concept_code` to match to the patients race. Since the race column in MIMIC is based on a string not a code this seems more appropriate. 
- Add a filter of `concept_class_id` on `mimic-race` to limit to the MIMIC race concepts.

Next, when building the person table I've made the following notable changes:
- Set `ethnicity_source_value` and `ethnicity_source_concept_id` to `NULL` since MIMIC only provides this information under race
- Set `race_concept_id` with the concept for the MIMIC race category except when the race is "HISPANIC OR LATINO". In this case `race_concept_id` is set to zero. 
- Set `ethnicity_concept_id` to the `target_concept_id` when the `domain_id` in the vocabulary is 'Ethnicity'. This currently only applies to races of "HISPANIC OR LATINO" which maps to Concept ID 38003563. MIMIC also has a number of race entries which contain the phrase "HISPANIC/LATINO" followed by more specific region information. In those cases we also map to 38003563. I keep the mapping of these two statements separate (even though they map to the same concept_id) for clarity. 

Another update creates the `person_id` in a deterministic manner instead of creating it based on `GENERATE_UUID()` which is random. This will allow for consistency with subsequent builds.

briangow committed:
update etl cdm_person per _delta vocab

atsvetkova-ody commented on etl/etl/cdm_person.sql:
```diff
@@ -115,62 +107,46 @@ SELECT
-    -- 
+    rc.race_first                   AS race_source_value,
+    map_rc.source_concept_id        AS race_source_concept_id,
+    CAST(NULL AS STRING)            AS ethnicity_source_value,
```
I would like to suggest to populate ethnicity_source_value and optionally ethnicity_source_concept_id when ethnicity_concept_id is not default. In fact, the only source field src_admissions.race provides values both for race_source_value and ethnicity_source_value, so it seems to me logical to populate both

briangow replied:
@atsvetkova-ody , thanks for the feedback! I've updated the code so that it handles ethnicity and race on a case by case basis. If the MIMIC term contains race and ethnicity information we map to `ethnicity` and `race` `source_value` and `source_concept_id`. If it is only applicable to one or the other, we only map to that respective one.

atsvetkova-ody commented on etl/etl/cdm_person.sql:
```diff
@@ -115,62 +107,46 @@ SELECT
+            WHEN map_rc.domain_id = 'Race' AND map_rc.source_code LIKE '%HISPANIC/LATINO%' THEN 38003563
             ELSE NULL
-        END, 0)                     AS ethnicity_concept_id,
+        END, 38003564)              AS ethnicity_concept_id,
```
I'm not sure if 38003564 "Not Hispanic Or Latino" is the best default value. Usually 0 is preferable to distinguish cases when a patient is explicitly not Hispanic and when a patient's ethnicity is unknown.

briangow replied:
Thanks, I've made this update.

atsvetkova-ody commented:
Beside the two comment above, the PR looks good!

somewordstoolate reviewed (approved):
Thanks for the update! This change fixed the issue I encountered during the MIMIC-IV v2.2 to OMOP conversion.

briangow commented:
@somewordstoolate , glad to hear that we have fixed an issue you've run into. We recently set the `dev` branch as the default since a few of us are actively doing development against it. This branch isn't ready for a full ETL build yet though. To minimize confusion, I've switched the default branch back to `main`. This is the branch we'd recommend using if you can't wait a bit longer for the work on `dev` to be finished.

briangow committed:
update subject_id obfuscation

briangow commented:
I've added another commit to this PR to use the obfuscation method introduced in #50 for `subject_id`.

briangow commented:
@atsvetkova-ody , could you clarify which comments you were referring to:

> Beside the two comment above, the PR looks good!

atsvetkova-ody commented:
> @atsvetkova-ody , could you clarify which comments you were referring to:
> 
> > Beside the two comment above, the PR looks good!

The comments to lines 121 and 128.
Oh, both are in "Pending" state, and there is an "eye" icon on them. You probably could not see it. Let me copy it here:

line 121
I'm not sure if 38003564 "Not Hispanic Or Latino" is the best default value. Usually 0 is preferable to distinguish cases when a patient is explicitly not Hispanic and when a patient's ethnicity is unknown.

line 128
I would like to suggest to populate ethnicity_source_value and optionally ethnicity_source_concept_id when ethnicity_concept_id is not default. In fact, the only source field src_admissions.race provides values both for race_source_value and ethnicity_source_value, so it seems to me logical to populate both

briangow commented:
Thanks for the feedback @atsvetkova-ody!

> line 121
I'm not sure if 38003564 "Not Hispanic Or Latino" is the best default value. Usually 0 is preferable to distinguish cases when a patient is explicitly not Hispanic and when a patient's ethnicity is unknown.

I had run this by colleagues a while back but looking at it now, I think I agree with you. I don't think we want to map to "Not Hispanic or Latino" by default. 

> line 128
I would like to suggest to populate ethnicity_source_value and optionally ethnicity_source_concept_id when ethnicity_concept_id is not default. In fact, the only source field src_admissions.race provides values both for race_source_value and ethnicity_source_value, so it seems to me logical to populate both

Yeah, since MIMIC only uses a `race` column to provide race and ethnicity this gets quite confusing. I'll review the best way to handle this. 

@p-talapova or @jshoughtaling , please let us know if you have any thoughts w/r/t these points.

briangow committed:
set ethnicity_concept_id to zero if unknown

briangow committed:
update ethnicity and race source value mapping

briangow committed:
update obf_id call to specify bits
